MDNSDiscoveryServiceOSGiTest unstable

https://ci.openhab.org/job/PR-openHAB-Core/1034/consoleText

```
[INFO] ------------------------------------------------------------------------
[INFO] Building openHAB Core :: Integration Tests :: Configuration mDNS Discovery Tests 2.5.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Deleting /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/target
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/src/main/resources
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Changes detected - recompiling the module!
[WARNING] /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/src/main/java/org/eclipse/smarthome/config/discovery/mdns/internal/MDNSDiscoveryServiceOSGiTest.java:[60,2243] Potential null pointer access: The variable configAdmin may be null at this location
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ org.openhab.core.config.discovery.mdns.tests ---
[WARNING] /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/itest.bndrun [0:0]: Unused Import-Package instructions: [org.osgi.framework.*] 
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] No tests to run.
[JENKINS] Recording test results
[INFO] 
[INFO] --- maven-jar-plugin:3.0.2:jar (default-jar) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Building jar: /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/target/org.openhab.core.config.discovery.mdns.tests-2.5.0-SNAPSHOT.jar
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (index) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] 
[INFO] --- bnd-indexer-maven-plugin:4.1.0:index (test-index) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] 
[INFO] --- bnd-testing-maven-plugin:4.1.0:testing (default) @ org.openhab.core.config.discovery.mdns.tests ---
[INFO] Matching glob *
16:37:31.374 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created thread pool 'safeCall' with size 5
16:37:31.574 [main] DEBUG org.eclipse.smarthome.core.internal.i18n.I18nProviderImpl - No language set, setting locale to 'null'.
16:37:31.574 [main] DEBUG org.eclipse.smarthome.core.internal.i18n.I18nProviderImpl - Error setting measurement system for value 'null'.
16:37:31.686 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'scheduler' of size 5
16:37:31.760 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created thread pool 'ConfigStatusService' with size 5
16:37:31.883 [main] DEBUG org.eclipse.smarthome.core.thing.internal.firmware.FirmwareUpdateServiceImpl - Modifying the configuration of the firmware update service.
16:37:31.884 [main] DEBUG org.eclipse.smarthome.core.thing.internal.firmware.FirmwareUpdateServiceImpl - Given time unit null is not supported. Will keep current configuration.
16:37:31.970 [main] DEBUG org.eclipse.smarthome.io.transport.mdns.internal.MDNSActivator - mDNS service has been started.
16:37:31.991 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'common' of size 5
16:37:32.017 [main] DEBUG javax.jmdns.impl.JmDNSImpl - JmDNS instance created
16:37:32.040 [main] DEBUG org.eclipse.smarthome.io.transport.mdns.internal.MDNSClientImpl - mDNS service has been started (JmDNS-/172.17.0.2 for IP 172.17.0.2)
16:37:32.042 [main] DEBUG org.eclipse.smarthome.core.common.ThreadPoolManager - Created scheduled thread pool 'discovery' of size 5
16:37:32.044 [main] DEBUG org.eclipse.smarthome.config.discovery.AbstractDiscoveryService - Background discovery for discovery service 'org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService' enabled.
16:37:33.010 [ESH-common-1] DEBUG org.eclipse.smarthome.core.net.NetUtil - added 1 network interfaces: [172.17.0.2/16]
16:37:33.011 [ESH-common-1] DEBUG org.eclipse.smarthome.core.net.NetUtil - removed 0 network interfaces: []
16:37:33.025 [safeCall-1] DEBUG org.eclipse.smarthome.io.transport.mdns.internal.MDNSClientImpl - ip address change: added [172.17.0.2/16], removed []
16:37:33.026 [safeCall-1] DEBUG javax.jmdns.impl.JmDNSImpl - Cancelling JmDNS: 
	---- Local Host -----
	local host info[JmDNS-/172-17-0-2.local., eth0:172.17.0.2, DNS: JmDNS-/172.17.0.2 [/172.17.0.2] state: probing 1 task: Prober(JmDNS-/172.17.0.2) state: probing 1]
	---- Services -----
	---- Types ----

	---- cache ----
	---- Service Collectors ----
	---- Service Listeners ----
16:37:33.027 [safeCall-1] DEBUG javax.jmdns.impl.JmDNSImpl - Canceling the timer
16:37:33.027 [safeCall-1] DEBUG javax.jmdns.impl.JmDNSImpl - unregisterAllServices()
16:37:33.029 [safeCall-1] DEBUG javax.jmdns.impl.JmDNSImpl - disposeServiceCollectors()
16:37:33.029 [safeCall-1] DEBUG javax.jmdns.impl.JmDNSImpl - Wait for JmDNS cancel: 
	---- Local Host -----
	local host info[JmDNS-/172-17-0-2.local., eth0:172.17.0.2, DNS: JmDNS-/172.17.0.2 [/172.17.0.2] state: canceling 1 task: Canceler(JmDNS-/172.17.0.2) state: canceling 1]
	---- Services -----
	---- Types ----

	---- cache ----
	---- Service Collectors ----
	---- Service Listeners ----
16:37:33.040 [JmDNS(JmDNS-/172.17.0.2).State.Timer] DEBUG javax.jmdns.impl.tasks.state.DNSStateTask - Canceler(JmDNS-/172.17.0.2).run() JmDNS canceling JmDNS-/172.17.0.2
16:37:33.048 [JmDNS(JmDNS-/172.17.0.2).State.Timer] DEBUG javax.jmdns.impl.tasks.state.DNSStateTask - Canceler(JmDNS-/172.17.0.2).run() JmDNS canceling #canceling 1
16:37:34.036 [JmDNS(JmDNS-/172.17.0.2).State.Timer] DEBUG javax.jmdns.impl.tasks.state.DNSStateTask - Canceler(JmDNS-/172.17.0.2).run() JmDNS canceling JmDNS-/172.17.0.2
16:37:34.036 [JmDNS(JmDNS-/172.17.0.2).State.Timer] DEBUG javax.jmdns.impl.tasks.state.DNSStateTask - Canceler(JmDNS-/172.17.0.2).run() JmDNS canceling #canceling 2
TEST testDynamicConfigurationOfBackgroundDiscovery(org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest) <<< ERROR: 
Expected: is <false>
     but: was <true>
java.lang.AssertionError: 
Expected: is <false>
     but: was <true>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest.lambda$1(MDNSDiscoveryServiceOSGiTest.java:53)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:152)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:120)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:84)
	at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest.testDynamicConfigurationOfBackgroundDiscovery(MDNSDiscoveryServiceOSGiTest.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:340)
	at aQute.junit.Activator.automatic(Activator.java:236)
	at aQute.junit.Activator.run(Activator.java:177)
	at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1377)
	at aQute.launcher.Launcher.run(Launcher.java:349)
	at aQute.launcher.Launcher.main(Launcher.java:149)
Tests run  : 1
Passed     : 0
Errors     : 1
Failures   : 0
[INFO] 1 Error(s)
[ERROR] Error   : Exit code remote process 1: java -cp /var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/target/test/itest/cnf/cache/4.1.0/org.openhab.core.bom.runtime-index/org.eclipse.osgi-3.13.100.v20180827-1536.jar:/var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/target/test/itest/cnf/cache/4.1.0/bnd-cache/biz.aQute.launcher/biz.aQute.launcher-4.1.0.jar -Dlauncher.properties="/var/jenkins_home/workspace/PR-openHAB-Core/itests/org.openhab.core.config.discovery.mdns.tests/target/test/itest/generated/launch6977449286556461461.properties" -ea aQute.launcher.Launcher
```

This is still a problem.

```
TEST testDynamicConfigurationOfBackgroundDiscovery(org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest) <<< ERROR: 
Expected: is <false>
     but: was <true>
java.lang.AssertionError: 
Expected: is <false>
     but: was <true>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest.lambda$1(MDNSDiscoveryServiceOSGiTest.java:53)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:152)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:120)
	at org.eclipse.smarthome.test.java.JavaTest.waitForAssert(JavaTest.java:84)
	at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryServiceOSGiTest.testDynamicConfigurationOfBackgroundDiscovery(MDNSDiscoveryServiceOSGiTest.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at aQute.junit.Activator.test(Activator.java:340)
	at aQute.junit.Activator.automatic(Activator.java:236)
	at aQute.junit.Activator.run(Activator.java:177)
	at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
	at aQute.launcher.Launcher.run(Launcher.java:352)
	at aQute.launcher.Launcher.main(Launcher.java:152)
```

--- a/itests/org.openhab.core.config.discovery.mdns.tests/src/main/java/org/eclipse/smarthome/config/discovery/mdns/internal/MDNSDiscoveryServiceOSGiTest.java
+++ b/itests/org.openhab.core.config.discovery.mdns.tests/src/main/java/org/eclipse/smarthome/config/discovery/mdns/internal/MDNSDiscoveryServiceOSGiTest.java
@@ -46,11 +46,13 @@ public class MDNSDiscoveryServiceOSGiTest extends JavaOSGiTest {
      */
     @Test
     public void testDynamicConfigurationOfBackgroundDiscovery() throws IOException {
-        setBackgroundDiscoveryViaConfigAdmin(true);
         waitForAssert(() -> assertThat(mdnsDiscoveryService.isBackgroundDiscoveryEnabled(), is(true)), 5000, 100);
 
         setBackgroundDiscoveryViaConfigAdmin(false);
         waitForAssert(() -> assertThat(mdnsDiscoveryService.isBackgroundDiscoveryEnabled(), is(false)), 5000, 100);
+
+        setBackgroundDiscoveryViaConfigAdmin(true);
+        waitForAssert(() -> assertThat(mdnsDiscoveryService.isBackgroundDiscoveryEnabled(), is(true)), 5000, 100);
     }
 
     private void setBackgroundDiscoveryViaConfigAdmin(boolean status) throws IOException {

